File: user-service/src/main/java/rs/edu/raf/mapper/UserMapper.java
package rs.edu.raf.mapper;

import org.springframework.stereotype.Component;
import rs.edu.raf.dto.LoggedUserDTO;
import rs.edu.raf.model.user.Employee;
import rs.edu.raf.model.user.Role;
import rs.edu.raf.model.user.User;

@Component
public class UserMapper {

    public LoggedUserDTO userToLoggedUserDTO(User user) {
        if(user instanceof Employee employee) {
            return new LoggedUserDTO(employee.getId(), employee.getUsername(), employee.getCompany().getId(), user.getRoles().stream().map(Role::getName).toList());
        }
        return new LoggedUserDTO(user.getId(), user.getUsername(), null, user.getRoles().stream().map(Role::getName).toList());
    }
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/mapper/UserMapper.kt
package org.example.userservicekotlin.mapper

import org.example.userservicekotlin.dto.LoggedUserDTO
import org.example.userservicekotlin.model.user.Employee
import org.example.userservicekotlin.model.user.User
import org.springframework.stereotype.Component

@Component
class UserMapper {
    fun userToLoggedUserDTO(user: User): LoggedUserDTO {
        if (user is Employee) {
            return LoggedUserDTO(
                user.id,
                user.username,
                user.company.id,
                user.roles.map{role->role.name}.toList()
            )
        }
        return LoggedUserDTO(
            user.id,
            user.username,
            null,
            user.roles.map {role ->role.name}.toList()
        )
    }
}